### PR TITLE
Change the Arrival messages to be nonempty

### PR DESCRIPTION
**Pull request by amaalali**
This changes the Arrival model to not allow the messages to be empty and removes logic that handles the empty messages which is not possible.

### CODE REVIEW

File: app/controllers/MovementsController.scala [hunk 1]
```diff
@@ -55,7 +56,7 @@ class MovementsController @Inject()(
 )(implicit ec: ExecutionContext)
     extends BackendController(cc) {
 
-  private val allMessageUnsent: Seq[MovementMessage] => Boolean =
+  private val allMessageUnsent: NonEmptyList[MovementMessage] => Boolean =
```
Comment (chris-ianson) [inline app/controllers/MovementsController.scala:59]: What's this used for?

File: app/models/Arrival.scala [hunk 2]
```diff
@@ -27,12 +30,30 @@ case class Arrival(
...
   updated: LocalDateTime,
-  messages: Seq[MovementMessage],
+  messages: NonEmptyList[MovementMessage],
   nextMessageCorrelationId: Int
 )
 
 object Arrival {
 
+  @deprecated("use NonEmptyList[MovementMessage] over Seq[MovementMessage]", "now")
+  def apply2(arrivalId: ArrivalId,
```
Comment (chris-ianson) [inline app/models/Arrival.scala:40]: How come we're marking this as deprecated and not just changing it to use a NonEmptyList?

Comment (amaalali) [reply app/models/Arrival.scala:40]: Oops this was while I was trying to get back to a compiling state, I'll get rid of that

File: app/controllers/MovementsController.scala (hunk 1 above)
Comment (amaalali) [reply app/controllers/MovementsController.scala:59]: This is the check that determines if a movement is will end up being created as a brand new or as part of a previous movement in which all messages (IE007) are unsubmitted

File: app/models/Arrival.scala (hunk 2 above)
Comment (amaalali) [reply app/models/Arrival.scala]: Removed

**Pull request merged**